### PR TITLE
Implement read operation (for readonly partition) - iOS

### PR DESCRIPTION
**Pull request by jackycodes**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

[AB#62453](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/62453)

### CODE REVIEW

File: appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m [hunk 1]
```diff
@@ -40,4 +50,24 @@ + (void)register {
...
+RCT_EXPORT_METHOD(read:(NSString *)documentId
+                  partition:(NSString *) partition
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject){
+    [MSData readDocumentWithID: documentId
+                  documentType:[MSDictionaryDocument class]
+                     partition: partition
+             completionHandler:^(MSDocumentWrapper *_Nonnull document) {
+                 NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+                 dict[kMSDeserializedValue] = document.deserializedValue.serializeToDictionary;
```
Comment (guperrot) [inline appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m]: Is this pseudo code the `serializeToDictionary`?

Comment (jackycodes) [reply appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m]: I'm using`serializeToDictionary` in  [MSSerializableDocument.h](https://github.com/microsoft/appcenter-sdk-apple/blob/develop/AppCenterData/AppCenterData/MSSerializableDocument.h)

File: appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m [hunk 2]
```diff
@@ -24,13 +24,23 @@
 #import <AppCenter/MSAppCenter.h>
 #import <AppCenterData/MSData.h>
 #import <AppCenterReactNativeShared/AppCenterReactNativeShared.h>
+#import <AppCenterData/MSDocumentWrapper.h>
```
Comment (dhei) [inline appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m]: sort imports

File: appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m (hunk 1 above)
Comment (jackycodes) [reply appcenter-data/ios/AppCenterReactNativeData/AppCenterReactNativeData.m]: Changed to `[document.deserializedValue serializeToDictionary]`

**Comment by dhei** [comment]
We will implement the error handling of `read` API in a separate PR.

**Pull request merged**